homeassistant/components/mqtt/climate.py

```diff
@@ -541,7 +593,7 @@ def handle_action_received(msg: ReceiveMessage) -> None:
         add_subscription(topics, CONF_ACTION_TOPIC, handle_action_received)
 
         @callback
-        def handle_temperature_received(
+        def handle_climate_attribute_received(
```

Comment:
The only line that seems to be de duplicated seems the log line, which is stale for `humidity`. May be we should just assign the attributes value in the specific handlers and dispose this one.

Reply:
Line 627 still shows `Could not parse temperature from`.
This is not applicable for humidity attributes. As I already commented it might be better not using an extra function for setting attributes. Directly setting them perhaps would be more directly and clear.

Reply:
Looks like I had forgotten to submit my pending comments, so you weren't able to see my comment. Can you take a look at my suggestion above?

Reply:
> Could not parse temperature_state_template from [...]

I think that is a good alternative!